api/v1/view_types.go

```diff
@@ -16,29 +16,54 @@ import (
 	"github.com/flanksource/incident-commander/api"
 )
 
+type ViewMergeStrategy string
+
+const (
+	ViewMergeStrategyLeft  ViewMergeStrategy = "left"
+	ViewMergeStrategyUnion ViewMergeStrategy = "union"
+)
+
+// ViewMergeSpec defines how to merge/join data from multiple queries
+type ViewMergeSpec struct {
+	// Strategy defines the merge strategy (left join or union)
+	// Default: union
+	// +kubebuilder:validation:Enum=left;union
+	// +kubebuilder:validation:Optional
+	Strategy ViewMergeStrategy `json:"strategy" yaml:"strategy"`
+
+	// Order defines the order of queries for joining
+	// +kubebuilder:validation:MinItems=1
+	Order []string `json:"order" yaml:"order"`
+}
+
+// PrometheusQuery defines a Prometheus query configuration
+type PrometheusQuery struct {
```

Comment:
I think we should extract this out into a generic duty.dataQuery package we can then add SQL / Clickhouse / Logs / HTTP that all return a generic table like structure

Reply:
https://github.com/flanksource/duty/pull/1477